python/src/deltachat/account.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import print_function
+import threading
+import requests
+from . import capi
+import deltachat
+from .capi import ffi
+
+
+class Account:
+    def __init__(self, db_path, logcallback=None):
+        self.dc_context = ctx = capi.lib.dc_context_new(
+                                  capi.lib.py_dc_callback,
+                                  capi.ffi.NULL, capi.ffi.NULL)
+        capi.lib.dc_open(ctx, db_path, capi.ffi.NULL)
+        self._logcallback = logcallback
+
+    def set_config(self, **kwargs):
+        for name, value in kwargs.items():
+            capi.lib.dc_set_config(self.dc_context, name, value)
+
+    def start(self):
+        deltachat.set_context_callback(self.dc_context, self.process_event)
+        capi.lib.dc_configure(self.dc_context)
+        self._threads = IOThreads(self.dc_context)
+        self._threads.start()
+
+    def shutdown(self):
+        deltachat.clear_context_callback(self.dc_context)
+        self._threads.stop(wait=False)
+        # XXX actually we'd like to wait but the smtp/imap
+        # interrupt idle calls do not seem to release the
+        # blocking call to smtp|imap idle. This means we
+        # also can't now close the database because the
+        # threads might still need it
+        # capi.lib.dc_close(self.dc_context)
+
+    def process_event(self, ctx, evt_name, data1, data2):
+        assert ctx == self.dc_context
+        if self._logcallback is not None:
+            self._logcallback((evt_name, data1, data2))
+        callname = evt_name[3:].lower()
+        method = getattr(self, callname, None)
```

Comment:
I'm not a fan of the lookup on self here.  Dependency injection being better than inheritance and all that.